Sources/_CryptoExtras/RSA/RSA+BlindSigning.swift

```diff
@@ -0,0 +1,414 @@
+    ///
+    /// - WARNING: Applications that require deterministic signatures can use the RSABSSA-SHA384-PSSZERO-Deterministic
+    /// variant, but only if their input messages have high entropy. Applications that use
+    /// RSABSSA-SHA384-PSSZERO-Deterministic SHOULD carefully analyze the security implications, taking into account
+    /// the possibility of adversarially generated signer keys as described in Section 7.3. When it is not clear whether
+    /// an application requires deterministic or randomized signatures, applications SHOULD use one of the variants with
+    /// randomized message preparation.
+    ///
+    /// - Seealso: [RFC 9474: RSABSSA Variants](https://www.rfc-editor.org/rfc/rfc9474.html#name-rsabssa-variants).
+    public static let RSABSSA_SHA384_PSSZERO_Deterministic = Self<SHA384>(padding: .PSSZERO, preparation: .identity)
+}
+
+extension _RSA.BlindSigning {
+    /// An input ready to be blinded, possibly prepended with random bytes.
+    ///
+    /// Users should not attempt to create values of this type; it is created and returned by the prepare operation.
+    public struct PreparedMessage {
+        var rawRepresentation: Data
+    }
+
+    /// The blinding inverse for a blinded messaeg, used to unblind a blind signature.
+    ///
+    /// Users should not attempt to create values of this type; it is created and returned by the blind operation.
+    public struct BlindInverse {
+        var rawRepresentation: Data
+    }
+}
+
+extension _RSA.BlindSigning.PrivateKey {
+    /// Generate a blind signature with the given key for a blinded message.
+    ///
+    /// - Parameter message: The blinded message to sign.
+    /// - Returns: A blind signature.
+    /// - Throws: If there is a failure producing the signature.
+    ///
+    /// - Seealso: [RFC 9474: BlindSign](https://www.rfc-editor.org/rfc/rfc9474.html#name-blindsign).
+    public func blindSignature<D: DataProtocol>(for message: D) throws -> _RSA.BlindSigning.BlindSignature {
+        try self.backing.blindSignature(for: message)
+    }
+}
+
+extension _RSA.BlindSigning.PublicKey {
+    /// Prepare a message to be signed using the blind signing protocol.
+    ///
+    /// - Parameter message: The message to be signed.
+    /// - Parameter parameters: Parameters used in the blind signing protocol.
+    /// - Returns: A preprared mesage, modified according to the parameters provided.
+    ///
+    /// - Seealso: [RFC 9474: Prepare](https://www.rfc-editor.org/rfc/rfc9474.html#name-prepare).
+    public func prepare<D: DataProtocol>(_ message: D) -> _RSA.BlindSigning.PreparedMessage {
+        switch self.parameters.preparation {
+        case .identity:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(message))
+        case .randomized:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(SystemRandomNumberGenerator.randomBytes(count: 32)) + message)
```

Comment:
This is a fairly inefficient construction. We're going to do the following:

1. Allocate a 32-byte array.
2. Attempt to append a message to it, which will require allocating a new Array.
3. Allocate a Data and copy that array into it.

A better algorithm is to allocate a Data, reserve 32 bytes + message.count, and then append the random bytes and the message. That will remove two allocations.

While we're here, can we add `@inlinable` on our generic functions?

Reply:
I've updated the data construction but we cannot make these `@inlinable` without some larger blast radius that would affect the existing RSA code.

Specifically because the backing key is private and there's a lot of very deliberate `private` and `fileprivate` layering which would all need to be diluted to make way for the `@usableFromInline`.

Should we defer this from this PR?